### ISSUE TITLE
location.mock attribute seems to be technically possible on iOS 15+ - can we have it in the library?

### ISSUE THREAD

**Issue by leobudima** [transistorsoft/cordova-background-geolocation-lt#1365]
## Your Environment
* Plugin version: 4.8.1
* Platform: iOS
* OS version: 15+
* Device manufacturer / model: Apple iPhone
* Cordova version (`cordova -v`): any
* Cordova platform version (`cordova platform ls`): any
* Plugin config provided to `#ready`: N/A

## Expected Behavior

Currently, `location.mock` attribute is available only for Android, for obvious and understandable reasons that Apple has previously not provided a direct way to determine it. 

Since iOS 15, the `isSimulatedBySoftware` instance property has been provided as a member of `CLLocationSourceInformation` class ( https://developer.apple.com/documentation/corelocation/cllocationsourceinformation/3861807-issimulatedbysoftware ) . 

Could the library be extended to support `mock` for iOS based on this information as well, please?

## Context

Our use-case involves crowdsourced drivers / couriers, and use of mock providers in order to game the system / customers is significant enough to affect the level of provided service, so we've developed defense mechanisms to prevent it, but can only cover a portion of the user base on Android. Supporting `mock` on iOS would help cover the rest. Thank you!

**Comment by christocracy** [comment]
>Plugin version: any / latest

This does not mean anything to me.  It's particularly meaningless in the future for others visiting this issue.  What is the **EXACT** version you're using?

**Comment by leobudima** [comment]
Hi Chris, 4.8.1 seems to be the latest version of the plugin, so I believe this is relevant for it.

**Comment by christocracy** [comment]
This has been implemented in the *private*, customers-only version.  

```
* Add new iOS 15 `CLLocation` attribute `Location.ellipsoidal_altitude` *The altitude as a height above the World Geodetic System 1984 (WGS84) ellipsoid, measured in meters*.  Android `Location.altitude` has always returned *ellipsoidal altutude*, so both `Location.altitude` and `Location.ellipsoidal_altitude` will return the same value.
```

While you are a customer, so have access to the private version, you have posted this in the public repo.

Are you not using installing the plugin from the *private* repo?

**Comment by christocracy** [comment]
Actually, there it is:  released here in the public version in [4.8.0](https://github.com/transistorsoft/cordova-background-geolocation-lt/blob/master/CHANGELOG.md#480--2022-06-21), this past *June 21*.

**Comment by christocracy** [comment]
And here it is in the `TSLocationManager.xcframework` source code:

<img width="871" alt="Screen Shot 2022-09-08 at 11 34 17 AM" src="https://user-images.githubusercontent.com/14999/189164408-61581b08-2873-4d81-9245-ff016f15e289.png">

**Comment by leobudima** [comment]
I've realized too late that I've posted to the public repo, and wanted to avoid notification chatter etc to you from closing and reopening there - apologies! 

Amazing about the feature - having it done before it iOS 15 was officially released just once again speaks towards your dedication and quality of the products.

As a side note, before posting this, I checked the documentation (https://transistorsoft.github.io/cordova-background-geolocation/interfaces/location.html#mock) and saw that it still says [Android only], as well as searched the Github organization for `isSimulatedBySoftware` which returned no results, from which I concluded that it's not here. Apologies for taking up your time and thanks for great and prompt support, @christocracy .
